Removes ID computer parts (Removes computer hardware)

JohnFulpWillard opened the pull request:
## About The Pull Request

Removes the last computer part in the game: ID parts
Because this is removed, I also removed all computer hardware in the game, and removed mentions of it in the game.
There is still 'hardware', as in Computer, Tablet, or Laptop.

Computers now all hold one ID slot by default, the only time a second ID was needed was to use the access of both at once, and for the ID modification application. This was now replaced with a new UI that only has one tab, one ID slot:

https://user-images.githubusercontent.com/53777086/202801939-151b783f-75c8-46bf-a6c5-1b57b0d0da8e.mp4

## Why It's Good For The Game

Computer hardware is finally dead 🦀 

## Changelog

:cl:
balance: All modular computers now only have one ID slot, and cannot be upgraded.
qol: The HoP's access application now only has one app, logging in will directly modify the ID that's in it, making it less confusing to swap back and forth.
/:cl:

JohnFulpWillard committed:
Removes card slots

Tablets now have 2 card slots by default.

JohnFulpWillard committed:
Deletes computer hardware entirely

Computer hardware is now gone.

JohnFulpWillard committed:
fix problems

JohnFulpWillard committed:
prettierx and unused import

JohnFulpWillard committed:
more fixes

Mothblocks commented:
I guess there's the HoP tablet app which complicates things, blegh.

I think maybe that should change to require you to click on the UI button with the ID in your hand or something.

Mothblocks commented on code/game/objects/items/cards_ids.dm:
```diff
@@ -959,43 +959,35 @@
 	update_label()
 
-/obj/item/card/id/advanced/proc/on_holding_card_slot_moved(obj/item/computer_hardware/card_slot/source, atom/old_loc, dir, forced)
+/obj/item/card/id/advanced/proc/on_holding_card_slot_moved(/obj/item/modular_computer/tablet/source, atom/old_loc, dir, forced)
```
Find-and-replace error, we don't use leading slash for vars

MTandi commented:
Isn't it better UX when you have only 1 ID slot and make attack with another ID swap the cards? Hop can use login/logout function. And you always know what you get when you alt click the tablet.

Mothblocks commented:
That's what I think

JohnFulpWillard commented:
It would remove the PDA's unique ability to show and use 2 IDs at once which is one of the reasons why I had liked tablets so much from the start

JohnFulpWillard committed:
i was running on 7% battery and rushed this, my bad

GuillaumePrata commented:
An option would be for left clicking with an ID to place on slot 1, right click on slot 2
Ctrl click to remove from slot 1, ctrl right click (do we even have that function?) for slot 2

I assume this would work as intended for players that expect to swap IDs while allowing "veteran" players to have a nice control on how they manage multi IDs.

MTandi commented:
> An option would be for left clicking with an ID to place on slot 1, right click on slot 2 Ctrl click to remove from slot 1, ctrl right click (do we even have that function?) for slot 2

This is so unintuitive. I needed 1 year to understand that to spill reagents you need to be in combat mode and do right click. And now when someone made the windows deconstructed with right clicks it's so annoying. Every time you try to deconstruct something you end up hitting it instead.

Please don't add more unnecessary things to right click.

GuillaumePrata commented:
> Every time you try to deconstruct something you end up hitting it instead.

The smart way would be to block attacking objects with tools if you are not in combat mode, so not really a right click issue and just an issue with the default way our system works.
We had this issue with too many things already and it was specially frustrating with light tubes as they would break.

Mothblocks commented:
thats the point of the wallets though john

timothymtorres commented:
I think mothblocks is right.  I had to add context screentips to tablets not too long ago and the 2 slots for IDs is clunky and not intuitive.  I would have also liked to make it so that a tablet can only have a pAI or a ID card inserted and not both but it was beyond the scope of the PR.

Also it seems like all the laptop stuff and gear never gets used.  Should probably consider removing it as some point.

MTandi commented:
> Also it seems like all the laptop stuff and gear never gets used. Should probably consider removing it as some point.

I wanted to make it possible to use laptop with usb cable for circuits so that you can come to something with a laptop and "hack" it. Still figuring out how the circuitry works though.

zxaber commented:
We could totally change the access program to do one or both of the following;
- Move to a "Change Disk" function where you authorize a change using your ID, and then insert the target ID to make the change`*`
- Use the `tap()` proc, so that you can authorize a change and then hit the target ID with your device to set it. This would need something extra for consoles though.

Then we wouldn't need two ID card slots at all.

`*This will lead to HoPs changing access and then handing over their own ID by mistake. For better or worse.`

JohnFulpWillard committed:
Removes secondary ID

PDA's can now only hold one ID, like old PDAs were able to.
The HoP's console now only has one ID inside of it at all times, you swap the IDs out to access the access changing stuff. This will probably be confusing for players at first, but I think it's easier to learn than the old UI.

JohnFulpWillard commented:
https://user-images.githubusercontent.com/53777086/202799618-1d30e657-032d-4f54-af45-cf033a8e4c28.mp4

New ID access app's UI
(Ignore the ID dropping on the console instead of in-hand)

JohnFulpWillard committed:
removes many mentions of hardware

Removes mentions of hardware in many places, as currently the only 'hardware' is differences between tablets, computers, and laptops.
I also moved computer_damage.dm to the rest of modular computer stuff because it was just 2 procs (that tbh should be removed)

anclandor commented:
Why are they still called modular computers if you've removed all the modules?

Fikou commented on code/modules/modular_computers/computers/item/computer.dm:
```diff
@@ -306,25 +235,61 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
+		return ..()
+
+	if(user)
+		if(!issilicon(user) && in_range(src, user))
```
you sure it works correctly? in the video the id card pops on the ground rather than in hands.

JohnFulpWillard replied:
![image](https://user-images.githubusercontent.com/53777086/202888239-1dc50340-3fef-45f5-992e-5a8492e427e4.png)

I had fixed it and tested it after I had made the video

JohnFulpWillard commented:
> Why are they still called modular computers if you've removed all the modules?

because they are modular

Mothblocks commented:
the modular part is the apps
